test/ParameterStoreBuildWrapperPluginTest.groovy

```diff
@@ -25,6 +33,236 @@ class ParameterStoreBuildWrapperPluginTest {
+
+                doReturn(environment).when(stage).getEnvironment()
+                doReturn(options).when(plugin).getParameterOptions(environment)
+                doReturn(expectedClosure).when(plugin).addParameterStoreBuildWrapper(options[0])
+
+                plugin.apply(stage)
+
+                verify(stage).decorate(TerraformEnvironmentStage.PLAN, expectedClosure)
+                verify(stage).decorate(TerraformEnvironmentStage.APPLY, expectedClosure)
+            }
+
+            @Test
+            void decorateTheTerraformEnvironmentStageWhenMultipleOptionsSet() {
+                def firstClosure                        = { -> }
+                def secondClosure                       = { -> }
+                String environment                      = "MyEnv"
+                List options                            = [[someKey: "someValue"], [someOtherKey: "someOtherValue"]]
+                TerraformEnvironmentStage stage         = mock(TerraformEnvironmentStage.class)
+                ParameterStoreBuildWrapperPlugin plugin = spy(new ParameterStoreBuildWrapperPlugin())
+
+                doReturn(environment).when(stage).getEnvironment()
+                doReturn(options).when(plugin).getParameterOptions(environment)
+                doReturn(firstClosure).when(plugin).addParameterStoreBuildWrapper(options[0])
+                doReturn(secondClosure).when(plugin).addParameterStoreBuildWrapper(options[1])
+
+                plugin.apply(stage)
+
+                verify(stage, times(2)).decorate(anyString(), eq(firstClosure))
+                verify(stage, times(2)).decorate(anyString(), eq(secondClosure))
+            }
+        }
+    }
+
+    class GetParameterOptions {
+        @After
+        public void reset() {
+            Jenkinsfile.instance = null
+            ParameterStoreBuildWrapperPlugin.reset()
+        }
+
+        private configureJenkins(Map config = [:]) {
+            Jenkinsfile.instance = mock(Jenkinsfile.class)
+            when(Jenkinsfile.instance.getStandardizedRepoSlug()).thenReturn(config.repoSlug)
+            when(Jenkinsfile.instance.getRepoName()).thenReturn(config.repoName ?: 'repo')
+            when(Jenkinsfile.instance.getOrganization()).thenReturn(config.organization ?: 'org')
+            when(Jenkinsfile.instance.getEnv()).thenReturn(config.env ?: [:])
+        }
+
+        @Test
+        void returnsNoOptionsWhenNotSet() {
+            String environment                      = "MyEnv"
+            Map option                              = [:]
+            List expected                           = [ option ]
+            ParameterStoreBuildWrapperPlugin plugin = spy(new ParameterStoreBuildWrapperPlugin())
+
+            doReturn(option).when(plugin).getEnvironmentParameterOptions(environment)
+
+            List actual = plugin.getParameterOptions(environment)
+
+            assertEquals(expected, actual)
+        }
```

Comment:
`returnsNoOptionsWhenNotSet` that name is weird, and doesn't match with what's being asserted. I think this is the result of reshuffling  behavior.  It's not that "NoOptions" are returned - it's that only environment-specific options are returned.